### ISSUE TITLE
URGENT: PDF build failing at fillin blank

### ISSUE THREAD

**Issue by rbeezer** [PreTeXtBook/pretext#1732]
Building sample book (noparts version) as PDF.  Getting a fatal error (PDF is incomplete).  Not a likely consequence of #1730 since reverting it does not provide a fix.  LaTeX is being created automatically from static version of Parson problem (Chapter 5).

A fix that will allow PDF-building to conclude (even if output is a bit subpar) would be very helpful. 

```tex
\begin{divisionexercise}{19}{}{}{x:exercise:fillin-string-integer}%
Complete the following line of a Python program so that it will declare an integer variable \mono{age} with an initial value of \mono{5}.%
\par
\fillintext{5} \mono{age = } \fillintext{5}\mono{;}%
\par\smallskip%
```

produces

```
! Leaders not followed by proper glue.
<to be read again> 
                   \hskip 
l.6816 \fillintext{5}
                      \mono{age = } \fillintext{5}\mono{;}%
```

**Comment by Alex-Jordan** [comment]
It seems that `\leaders` does not like to immediately follow a `\par`. Not even with the `\null` in front.

I put a `\rule{0pt}{0pt}` at the front of the macro, and that gets past this. Is that the kind of subpar solution that you'd like for now? Or should I look for a less hack-ish solution? (I will look either way, but what's the urgency on a non-hackish solution.)

**Comment by rbeezer** [comment]
So, start of the line, and start of the par?

Hack is good for now, just want the build to complete.  Thanks for the urgency.

**Comment by Alex-Jordan** [comment]
Another mystery with this. Note in #19 that some answer blanks have no underline. Screenshot from something I'm testing, but I have been seeing this with many variants trying to address the start-of-par issue.

<img width="664" alt="Screen Shot 2022-06-07 at 5 16 44 PM" src="https://user-images.githubusercontent.com/4732672/172505203-6e3ac4ac-94a2-4013-b6ee-1a5024304c82.png">

**Comment by Alex-Jordan** [comment]
I can recover the missing underlines by using another `\null` at the *end* of the fillin.

**Comment by rbeezer** [comment]
Go for it!  (Cooking now...)

**Comment by rbeezer** [comment]
I sometimes need to zoom-in in my PDF viewer to see "missing" underlines.  Have you checked that behavior?

**Comment by rbeezer** [comment]
Closed via  #1734

**Comment by Alex-Jordan** [comment]
Yeah, the missing underlines was not a zoom issue. When it was immediately followed by the `\mono` command without something in between, that triggered this symptom. No idea why. But you can see the missing underlines preceding the semicolons in the screenshot. Some mystery with "leaders" that maybe Michael would be able to explain.
